provisioner/ansible/provisioner.go

```diff
@@ -382,6 +391,21 @@ func validateFileConfig(name string, config string, req bool) error {
 	return nil
 }
 
+func validateDirectoryConfig(name string, config string, req bool) error {
```

Comment:
It's there to mirror `validateFileConfig`, which is used to validate the state of different files required by different options.  In those cases, a different value is passed into `config` for each different option.  I suppose the `config` param in `validateDirectoryConfig` falls under YAGNI maybe, since that function is only used once ever (in validating `inventory_directory`.  Given that, I could get rid of `validateDiretoryConfig` entirely and inline the whole thing, I guess; I was trying to make the validation of the `inventory_directory` setting consistent with the validation of the other settings that depended on stuff in the filesystem.